File: StockSpan.java
public ArrayList<Integer> calculateSpan(int[] arr) {
        // write code here
        Stack<Integer> st = new Stack<>();
        Integer[] res = new Integer[arr.length];
        res[0] = 1;
        for(int i = 0; i < arr.length; ++i){
            if(st.isEmpty()){
                st.push(i);
            }
            else{
                while(!st.isEmpty() && arr[i] >= arr[st.peek()]){
                    st.pop();
                }
                
                if(st.isEmpty()){
                    res[i] = i + 1;
                }
                else{
                    res[i] = i - st.peek();
                }
                
                st.push(i);
            }
        }
        return new ArrayList<>(Arrays.asList(res));
        
    }

File: StockSpan.cpp
    vector<int> calculateSpan(vector<int>& arr) {
        // write code here
        std::stack<int> st;
        std::vector<int> res(arr.size());
        res[0] = 1;
        
        for (size_t i = 0; i < arr.size(); ++i) {
            while (!st.empty() && arr[i] >= arr[st.top()]) {
                st.pop();
            }
            
            if (st.empty()) {
                res[i] = i + 1;
            } else {
                res[i] = i - st.top();
            }
            
            st.push(i);
        }
        
        return res;
    }
